src/components/Utility/FeeSelector.vue

```diff
@@ -144,11 +116,38 @@ export default {
     return {
       chosenFee: "normal",
       useCustomFee: false,
-      customFee: 30,
-      data: ["cheapest", "slow", "normal", "fast"]
+      customFee: 30
     };
   },
-  computed: {},
+  computed: {
+    isDisabled() {
+      return (
+        this.fee.fast.total <= 0 ||
+        this.fee.fast.total === "--" ||
+        this.fee.fast.total === "N/A" ||
+        this.disabled
+      );
+    },
+    recommendedFees() {
+      if (this.isDisabled) {
+        return ["cheapest", "slow", "normal", "fast"];
+      }
+      const intervals = [];
+      if (this.fee.cheapest && !this.fee.cheapest.error) {
+        intervals.push("cheapest");
+      }
+      if (this.fee.slow && !this.fee.slow.error) {
+        intervals.push("slow");
+      }
+      if (this.fee.normal && !this.fee.normal.error) {
+        intervals.push("normal");
+      }
+      if (this.fee.fast && !this.fee.fast.error) {
+        intervals.push("fast");
+      }
+      return intervals;
+    }
+  },
```

Comment:
Maybe this could be optimised to:

```suggestion
  recommendedFees() {
    const intervals = ["cheapest", "slow", "normal", "fast"];
    if (this.isDisabled) {
      return intervals;
    }
    return intervals.reduce((intervals, interval) => {
      if (this.fee[interval] && !this.fee[interval].error) {
        intervals.push(interval);
      }
    }, []);
  }
```

Reply:
Yeah, I hate it too, but we can't guarantee if the order would always be "cheapest", "slow", "normal" and "fast" (to show them from left-to-right on the slider).

Reply:
Tbh now I look back at both of them, I find the procedural method more readable and there's not that much code duplication.

Feel free to ignore this if you find the functional implementation less readable.

Reply:
>Yeah, I hate it too, but we can't guarantee if the order would always be "cheapest", "slow", "normal" and "fast" (to show them from left-to-right on the slider).

Arrays in JavaScript are ordered, they will always be looped over in that exact order.

Reply:
> Arrays in JavaScript are ordered, they will always be looped over in that exact order.

Ohh my bad, I didn't carefully read your code suggestion. I thought you were looping object keys of the `fee` object (which could be unordered), but yeah, looping over the `intervals` array would work. 

Agree with you on the current code's readability, we can stick to it.